http/1.7/modules/ROOT/pages/http-polling-source-ref.adoc

```diff
@@ -0,0 +1,300 @@
+</flow>
+----
+
+* The `watermarkExpression` has the `watermark` placeholder that retrieves the watermark from the property `wm` from the response.
+* In the first polling iteration, there are three executions of the flow, one execution with `{name: 'Eze'}`, another execution with `{name: 'Fabi'}` and the last execution with `{name: 'Sofi'}`.
+* In the second polling iteration, that occurs 5 minutes later, there are 2 executions, one execution with `{name: 'Euge'}` and another execution with `{name: 'Juli'}`).
+* In the third polling iteration, that also occurs 5 minutes later, there are two executions of the flow, one execution with `{name: 'Pablo'}` and another one with `{name: Martin}`).
+* Afterwards, there will be continue polling iterations but because the results are empty, there are no flow executions.
+
+The following XML configuration example shows how to extract the watermark value from each item, and then use the value in the query parameters. The behavior depends completely on how the HTTP server uses the watermarking value:
+
+[source,xml,linenums]
+----
+<flow name="watermarkIntoQueryParamsListenerFlow">
+    <http:listener config-ref="watermarkListenerConfig" path="/watermark-into-query"/>
+    <choice>
+        <when expression="#[attributes.queryParams.watermark == '0']">
+            <set-payload value="#[output application/json --- {'items': [{'name': 'Rodro', 'wm': 1}, {'name': 'Steve', 'wm': 2}, {'name': 'Juan', 'wm': 3}]}]"/>
+        </when>
+        <when expression="#[attributes.queryParams.watermark == '2']">
+            <set-payload value="#[output application/json --- {'items': [{'name': 'Axel', 'wm': 4}, {'name': 'Mariano', 'wm': 5}]}]"/>
+        </when>
+        <when expression="#[attributes.queryParams.watermark == '5']">
+            <set-payload value="#[output application/json --- {'items': [{'name': 'Ivan', 'wm': 6}, {'name': 'Hyeran', 'wm': 7}]}]"/>
+        </when>
+        <otherwise>
+            <set-payload value="#[output application/json --- {'items': []}]"/>
+        </otherwise>
+    </choice>
+</flow>
+
+<flow name="watermarkIntoQueryParamsPollingFlow">
+    <http:polling-source config-ref="watermarkRequestConfig" path="watermark-into-query"
+        splitExpression="#[payload.items]" watermarkExpression="#[item.wm]">
+        <scheduling-strategy>
+            <fixed-frequency frequency="1" timeUnit="SECONDS"/>
+        </scheduling-strategy>
+        <http:polling-request-query-params >
+            <http:polling-request-query-param key="watermark" value="#[watermark default 0]" />
+        </http:polling-request-query-params>
+    </http:polling-source>
+    <logger message="#[payload]"/>
+</flow>
+----
+
+== Configure Idempotency Expression
+
+To avoid the concurrent execution of two flows with the same payload identification, configure idempotency for the HTTP *Polling Source* event source. The configuration ensures that payloads with the same ID are processed not concurrently.
```

Comment:
```suggestion
To avoid the concurrent execution of two flows with the same payload identification, configure idempotency for the HTTP *Polling Source* event source. This configuration ensures that payloads with the same ID are not processed concurrently.
```